src/svg/qsvggenerator.cpp

```diff
@@ -997,8 +1007,37 @@ void QSvgPaintEngine::updateState(const QPaintEngineState &state)
     // close old state and start a new one...
     if (d->afterFirstUpdate)
         *d->stream << "</g>\n\n";
+    
+    QPainter* p = painter();
+    if (p->hasClipping()) {
+        QPainterPath painter_path = p->clipPathF();
+
+        if (painter_path.elementCount() > 0) {
+            QString clip_path = createPath(painter_path);
+
+            bool path_does_not_exist = 0 == clip_path_to_id.count(clip_path);
```

Comment:
Consider removing this variable and just moving it to the if brackets, since it's only used once

Reply:
I intentionally turned it back into a variable since I personally find it far easier to read, like a sentence "If path does not exist" rather than having to mentally notice what is happening within the other function calls.

Reply:
Yeah I take your point, but what if it read `if(clip_path_to_id.contains(clip_path)` or something like that? Not a huge issue to have one extra boolean on the stack of course

Reply:
I would have thought a single-use variable like this would be optimized away at compile-time, so it only enhanced readability.

Reply:
I did not want to risk oversimplification when it came to the original Qt code and simply let it how it was.